rpcs3/Emu/RSX/Common/texture_cache.h

```diff
@@ -10,328 +11,106 @@ extern u64 get_system_time();
 
 namespace rsx
 {
-	enum texture_create_flags
-	{
-		default_component_order = 0,
-		native_component_order = 1,
-		swapped_native_component_order = 2,
-	};
-
-	enum memory_read_flags
-	{
-		flush_always = 0,
-		flush_once = 1
-	};
-
-	struct typeless_xfer
-	{
-		bool src_is_typeless = false;
-		bool dst_is_typeless = false;
-		bool src_is_depth = false;
-		bool dst_is_depth = false;
-		u32 src_gcm_format = 0;
-		u32 dst_gcm_format = 0;
-		f32 src_scaling_hint = 1.f;
-		f32 dst_scaling_hint = 1.f;
-		texture_upload_context src_context = texture_upload_context::blit_engine_src;
-		texture_upload_context dst_context = texture_upload_context::blit_engine_dst;
-
-		void analyse()
-		{
-			if (src_is_typeless && dst_is_typeless)
-			{
-				if (src_scaling_hint == dst_scaling_hint &&
-					src_scaling_hint != 1.f)
-				{
-					if (src_is_depth == dst_is_depth)
-					{
-						src_is_typeless = dst_is_typeless = false;
-						src_scaling_hint = dst_scaling_hint = 1.f;
-					}
-				}
-			}
-		}
-	};
-
-	struct cached_texture_section : public rsx::buffered_section
+	template <typename commandbuffer_type, typename section_storage_type, typename image_resource_type, typename image_view_type, typename image_storage_type, typename texture_format>
```

Comment:
Please use CamelCase for template parameter names. And keep names  shorter.

Reply:
I used exactly the same names as were already used inside the texture_cache.h before my refactor, I don't think it makes sense to change them now.

Reply:
I guess so, but this is just template parameters, they are only visible within the class, and most the code is considered changed.